### PR TITLE
feature/#1128 - faster MapTileCache management

### PR DESCRIPTION
**Pull request by monsieurtanuki**
Removed class:
* `LRUMapTileCache`: already deprecated class, removed _en route_

Impacted classes:
* `IConfigurationProvider`: impacted the removal of `LRUMapTileCache` in comments; gently refactored
* `MapTileCache`: new `boolean mAutoEnsureCapacity` and its getter; optimized method `garbageCollection` using `mAutoEnsureCapacity`; method `ensureCapacity` now returns a `boolean` ("actually changed the capacity?"); optimized method `shouldKeepTile`
* `MapTilePreCache`: added a little optimization in method `fill`; fixed typos
* `MapTileProviderBasic`: set the new `MapTileCache.mAutoEnsureCapacity` to `true`

### CODE REVIEW

**Comment by spyhunter99** [comment]
aside from the one comment, nothing pops out as a potential issue. I'll pull down the branch and give it a whirl.

**Comment by monsieurtanuki** [comment]
> aside from the one comment

@spyhunter99 ???

File: osmdroid-android/src/main/java/org/osmdroid/tileprovider/MapTilePreCache.java [hunk 1]
```diff
@@ -78,11 +92,11 @@ private void refresh() {
     private long next() {
         while(true) {
             final long index;
-            synchronized (mTiles) {
-                if (mTilesIndex >= mTiles.getSize()) {
+            synchronized (mTileAreas) {
+                if(!mTileIndices.hasNext()) {
```
[the remark this answers is not in the document]
Comment (monsieurtanuki) [reply osmdroid-android/src/main/java/org/osmdroid/tileprovider/MapTilePreCache.java]: `mTileAreas` is refreshed in a `synchronized (mTileAreas)` statement, along with `mTileIndices` (`= mTileAreas.iterator();`), inside method `refresh`.
`mTileIndices` is accessed in a `synchronized (mTileAreas)` statement, inside method `next`.
We only refresh or access `mTileIndices` inside `synchronized (mTileAreas)` statements.

[the remark this answers is not in the document]
Comment (monsieurtanuki) [reply osmdroid-android/src/main/java/org/osmdroid/tileprovider/MapTilePreCache.java]: I assume that synchronizing on `mTileAreas` every time I access `mTileIndices` (input or output) is safe for `mTileIndices`. I don't need an extra `synchronized(mTileIndices)` step.

And I do need to use `synchronized(mTileAreas)` in `refresh`, therefore I cannot replace it with `synchronized(mTileIndices)`.

[the remark this answers is not in the document]
Comment (spyhunter99) [reply osmdroid-android/src/main/java/org/osmdroid/tileprovider/MapTilePreCache.java]: ok not looking to start a debate. i just saw the diff and thought it that looked like it was possible to invalidate the enumerator/iterator if the collection changes from an outside thread. Usually you synchronize on the collection, then iterate. Common concurrency issue. If it's not an issue here, then merge it

[the remark this answers is not in the document]
Comment (monsieurtanuki) [reply osmdroid-android/src/main/java/org/osmdroid/tileprovider/MapTilePreCache.java]: I read the code again and I'm confident.

**Pull request merged**